src/app/lib/utilities/ichefURL/index.js

```diff
@@ -2,28 +2,27 @@
 const WEBP_ORIGIN_CODES = ['cpsdevpb', 'cpsprodpb'];
 
 const buildPlaceholderSrc = (src, resolution) => {
-  const parts = src.split('/');
-  const [domain, media, imgService, width, ...extraParts] = parts;
-  const definedWidth = width.replace('$width', resolution);
+  // return standard img url pips
+  if (src.includes('urn:')) return src;
+  const noProtocolSrc = src.replace('https://', '');
+  const parts = noProtocolSrc.split('/');
+  const [domain, media, imgService, ...extraParts] = parts;
+  // shift is used to delete the old resolution.
+  extraParts.shift();
```

Comment:
are you removing the first item of the extraParts array? mutating data with methods like shift or pop is usually not a good idea. A quick google on the issue https://jaketrent.com/post/remove-array-element-without-mutating

Slice does not mutate the original array.

instead of
```js
extraParts.shift();
```

you should do something like

```js
const newParts = extraParts.slice(1)
```

on a side-note, these variable names could be better

Reply:
Thanks for the resources. I changed it to slice

Reply:
I tried to make the variables more readable